sridha2412/assignment2/q1.py

```diff
@@ -0,0 +1,36 @@
+# print ancestors of given key in a binary tree
+
+# node class`
+class Node:
+    # constuctor to initlalize a node object
+    def __init__(self, data):
+        self.data = data
+        self.left = None
+        self.right = None
+
+# if key is present, print ancestors else return false
+def print_ancestors(root, key):
```

Comment:
Please document the functions signatures with Python's type annotations.
1. this makes it easier for the reader,
2. this makes you think about what makes sense for good types for your module.

I suggest you split the print_ancestors function in two:
1. one function that returns the list of ancestors,
2. one function that simply prints it.

The advantage of having a function that returns a list is that it is much more useful, instead of being scoped to simply printing. 

Even in our limited use case, it is much easier to do unit test on a list return value than on a values being printed. Testing that function prints what we expect is possible but less straightforward.